Example/Swift/ViewController.swift

```diff
@@ -188,31 +170,20 @@ class ViewController: UIViewController, MGLMapViewDelegate, AVSpeechSynthesizerD
         }
     }
     
-    func speak(_ text: String) {
-        let utterance = AVSpeechUtterance(string: text)
-        speechSynth.delegate = self
-        speechSynth.speak(utterance)
-    }
-    
     func startNavigation(_ route: Route) {
-        let camera = mapView.camera
-        camera.pitch = 40
-        mapView.setCamera(camera, animated: false)
-        mapView.userTrackingMode = .followWithCourse
-        navigation = RouteController(route: route)
-        navigation?.resume()
-    }
-    
-    func endNavigation() {
-        instructionView.isHidden = true
-        toggleNavigationButton.isHidden = true
-        howToBeginLabel.isHidden = false
-        mapView.userTrackingMode = .none
-        let camera = mapView.camera
-        camera.pitch = 0
-        mapView.setCamera(camera, animated: true)
-        removeRoutesFromMap()
-        navigation?.suspend()
+        // Pass through a
+        // 1. the route the user will take
+        // 2. A `Directions` class, used for rerouting.
+        let controller = NavigationUI.instantiate(route: route, directions: directions)
```

Comment:
I first added the NavigationUI to handle the theme only, but because of the way Pulley is set up with designated initializer which I can't override conveniently, a factory method was the most appropriate I could come up with. The terminology is inherited from
```swift
UIStoryboard.instantiateInitialViewController()
UIStoryboard.instantiateViewController(withIdentifier:)
```
That being said, I agree with you, I'd prefer an init method on RouteViewController. The problem is supporting initializing programmatically and through storyboard without using a factory method. I guess I could fork Pulley but the intention is to replace it eventually.